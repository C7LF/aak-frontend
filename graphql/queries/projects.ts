import { gql } from '@apollo/client';

export const GET_PROJECTS = gql`
  query {
    projects {
      id
      slug
      title
      image {
        url
        alternativeText
        formats
      }
    }
  }
`;

export const GET_PROJECT = gql`
  query GetProjectBySlug($slug: String!) {
    projects(where: { slug: $slug }) {
      id
      slug
      title
      heroContent
      content
      image {
        url
        alternativeText
        formats
      }
      fancyGallery {
        images {
          url
          width
          height
          caption
          alternativeText
          formats
        }
      }
    }
  }
`;
