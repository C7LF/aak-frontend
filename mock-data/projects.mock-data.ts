import { SingleProject } from '@models/project.model';

export const ProjectsMockData: SingleProject[] = [
  {
    id: 1,
    title: 'First Post!!',
    slug: 'wow',
    content:
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut ut odio tempor, auctor tellus id, rutrum est. Donec purus massa, blandit eu efficitur nec, sodales suscipit libero. Integer varius magna ornare tortor consectetur sagittis. Aliquam nulla massa, imperdiet ac sem nec, molestie volutpat lorem. Curabitur quam urna, bibendum dictum lacinia at, congue in tellus. Praesent sed dolor sed dolor feugiat auctor. Integer cursus ultricies magna. Nam quis tempor tortor, eget cursus dui. Nam volutpat, arcu eu molestie vehicula, dolor turpis vestibulum libero, et dignissim ipsum odio eu ipsum. Vestibulum elementum tincidunt viverra. Nam elementum ex vitae orci mattis, quis ultrices mauris venenatis. Sed ut ullamcorper velit. Nullam ac eleifend enim. Suspendisse consectetur libero augue, id dignissim diam volutpat et. Aliquam ac dapibus ante, a consectetur turpis. Curabitur in vulputate eros. Praesent posuere, nunc ut bibendum semper, enim metus scelerisque ipsum, ac tempus neque ligula eget dui. Vestibulum a lacus dolor. Vivamus eget malesuada justo. Pellentesque sit amet dignissim lectus, eu ultrices diam. Nunc volutpat a ante eu placerat. Praesent vestibulum vehicula quam, quis condimentum ligula ornare ut. Aenean laoreet eros mauris, rutrum ultrices enim mollis id. Nulla tincidunt, turpis vitae egestas facilisis, magna metus sodales dolor, nec tincidunt ligula ipsum eu ligula. Etiam placerat tincidunt tellus, et pulvinar enim luctus sed.',
    published_at: '2020-12-28T19:46:55.914Z',
    created_at: '2020-12-28T19:46:19.193Z',
    updated_at: '2021-01-02T20:58:35.687Z',
    image: {
      id: 1,
      name: 'camaro at newark.jpg',
      alternativeText: '',
      caption: '',
      width: 639,
      height: 479,
      formats: {
        thumbnail: {
          name: 'thumbnail_camaro at newark.jpg',
          hash: 'thumbnail_camaro_at_newark_66a0558cb0',
          ext: '.jpg',
          mime: 'image/jpeg',
          width: 208,
          height: 156,
          size: 11.87,
          path: null,
          url: '/uploads/thumbnail_camaro_at_newark_66a0558cb0.jpg',
        },
        small: {
          name: 'small_camaro at newark.jpg',
          hash: 'small_camaro_at_newark_66a0558cb0',
          ext: '.jpg',
          mime: 'image/jpeg',
          width: 500,
          height: 375,
          size: 57.91,
          path: null,
          url: '/uploads/small_camaro_at_newark_66a0558cb0.jpg',
        },
        large: {
          name: 'large_camaro at newark.jpg',
          hash: 'large_camaro_at_newark_66a0558cb0',
          ext: '.jpg',
          mime: 'image/jpeg',
          width: 500,
          height: 375,
          size: 57.91,
          path: null,
          url: '/uploads/large_camaro_at_newark_66a0558cb0.jpg',
        },
      },
      hash: 'camaro_at_newark_66a0558cb0',
      ext: '.jpg',
      mime: 'image/jpeg',
      size: 93.49,
      url: '/uploads/camaro_at_newark_66a0558cb0.jpg',
      previewUrl: null,
      provider: 'local',
      provider_metadata: null,
      created_at: '2021-01-02T20:58:15.625Z',
      updated_at: '2021-01-02T20:58:15.636Z',
    },
  },
];
