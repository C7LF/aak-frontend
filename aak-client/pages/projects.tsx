import React from 'react';

import { GetStaticProps } from 'next';

import { Footer, Gallery, NavBar } from '@components';
import { GET_PROJECTS } from '@graphql/queries/projects';
import client from '@lib/apollo';
import { SingleProject } from '@models/project.model';

interface ProjectsProps {
  projects: SingleProject[];
}

export const getStaticProps: GetStaticProps = async () => {
  const { data } = await client.query({
    query: GET_PROJECTS,
  });

  return {
    props: {
      projects: data.projects,
    },
    revalidate: 1,
  };
};

const Projects: React.FC<ProjectsProps> = ({ projects }) => {
  return (
    <>
      <NavBar />
      <div className="bg-gray-50 py-24 pt-32 md:pt-44 mb-16">
        <div className="container mx-auto">
          <h1 className="text-4xl font-bold text-gray-800">PROJECTS</h1>
          <div className="border border-b-4 border-gray-400 w-16 ml-0.5 my-6"></div>
          <p className="text-xl font-light">
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras
            maximus mi nec sem sodales, eget pellentesque neque pulvinar.
            Phasellus sollicitudin erat sit amet tempus gravida. Nam in ex
            molestie. Cras a tellus fringilla, euismod nibh non, vehicula leo.
            Vestibulum tempor pharetra arcu sed rutrum. Donec efficitur aliquam
            dolor nec suscipit. Vivamus dui libero.
          </p>
        </div>
      </div>

      <Gallery items={projects} />
      <Footer />
    </>
  );
};

export default Projects;
