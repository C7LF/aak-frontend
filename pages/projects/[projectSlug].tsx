import React from 'react';

import { GetStaticPaths, GetStaticProps } from 'next';
import Image from 'next/image';

import { Layout, FancyGallery, Breadcrumbs } from '@components';
import { GET_PROJECT, GET_PROJECTS } from '@graphql/queries/projects';
import client from '@lib/apollo';
import { SingleProject } from '@models/project.model';

interface ProjectProps {
  project: SingleProject;
}

export const getStaticProps: GetStaticProps = async (context) => {
  const slug = context.params.projectSlug;

  const { data } = await client.query({
    query: GET_PROJECT,
    variables: { slug: slug.toString() },
  });

  return {
    props: {
      project: data.projects[0],
    },
    revalidate: 1,
  };
};

export const getStaticPaths: GetStaticPaths = async () => {
  const { data } = await client.query({
    query: GET_PROJECTS,
  });

  const paths = data.projects.map((project: SingleProject) => ({
    params: { projectSlug: project.slug.toString() },
  }));

  return {
    paths,
    fallback: false,
  };
};

const newLineText = (text: string) => {
  return text.split('\n').map((str: string, i) => (
    <p key={i} className="mt-8">
      {str}
    </p>
  ));
};

const Project: React.FC<ProjectProps> = ({ project }) => {
  if (project) {
    return (
      <Layout varient="full">
        <div className="container mx-auto">
          <Breadcrumbs />
        </div>
        <div className="md:flex">
          <div className="md:w-1/2 h-pi bg-gray-100">
            <div className="container inset-center mt-20">
              <div className="md:w-1/2 md:pr-12">
                <h1 className="text-5xl font-semibold mb-2">{project.title}</h1>
                <p className="font-light text-lg">{project.heroContent}</p>
              </div>
            </div>
          </div>
          <div className="md:w-1/2 h-pi relative">
            <Image
              src={`${process.env.NEXT_PUBLIC_API_URL}${project.image.url}`}
              alt={project.image.alternativeText}
              loading="lazy"
              layout="fill"
              objectFit="cover"
            />
          </div>
        </div>
        <div className="container mx-auto relative">
          {newLineText(project.content)}
          {project.fancyGallery && (
            <FancyGallery fancyItems={project.fancyGallery} />
          )}
        </div>
      </Layout>
    );
  }
};

export default Project;
