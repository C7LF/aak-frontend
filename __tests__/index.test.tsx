import React from 'react';

import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import { ProjectsMockData } from '@mock-data/projects.mock-data';

import Home from '../pages/index';

describe('Pages', () => {
  describe('Home', () => {
    it('it renders correctly', () => {
      const tree = renderer
        .create(<Home projects={ProjectsMockData} />)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('Home component should be defined', () => {
      const wrap = shallow(<Home projects={ProjectsMockData} />);
      expect(wrap).toBeDefined();
    });
  });
});
