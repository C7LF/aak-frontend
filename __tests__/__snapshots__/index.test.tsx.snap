// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pages Home it renders correctly 1`] = `
Array [
  <div
    className=" fixed z-20 md:hidden w-full transition-all duration-300 ease-in-out top-0"
  >
    <div
      className="container flex mx-auto py-5 text-black transition-all duration-300 ease-in-out"
    >
      <svg
        className="w-8 md:hidden mr-6"
        fill="none"
        onClick={[Function]}
        stroke="currentColor"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M4 6h16M4 12h16M4 18h16"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
        />
      </svg>
      <img
        alt="andy's american kustoms logo"
        className="mr-10 w-24"
        src="test-file-stub"
      />
    </div>
  </div>,
  <div
    className="transition-all duration-300 ease-in-out overflow-hidden w-full bg-white fixed top-0 z-10 md:py-5 md:mt-0 h-0 md:h-auto"
  >
    <div
      className="container mx-auto"
    >
      <nav
        className="flex flex-col md:flex-row"
      >
        <img
          alt="andy's american kustoms logo"
          className="mr-10 w-28 hidden md:block"
          src="test-file-stub"
        />
        <ul
          className="md:flex mt-16 md:mt-0 items-center pt-4 md:pt-0 uppercase font-medium"
        >
          <a
            className="md:p-4 py-3 px-0 block border-b-2 border-transparent hover:text-red-700"
            href="/"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            Home
          </a>
          <a
            className="md:p-4 py-3 px-0 block border-b-2 border-transparent hover:text-red-700"
            href="/about"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            About
          </a>
          <a
            className="md:p-4 py-3 px-0 block border-b-2 border-transparent hover:text-red-700"
            href="/projects"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            Projects
          </a>
          <a
            className="md:p-4 py-3 px-0 block border-b-2 border-transparent hover:text-red-700"
            href="/contact"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            Contact
          </a>
        </ul>
      </nav>
    </div>
  </div>,
  <main>
    <div
      className="top h-screen flex"
    >
      <div
        className="left rad-red transform md:-skew-x-6 h-full w-full lg:w-3/4 xl:w-1/2 overflow-x-hidden md:-ml-20 flex justify-center md:justify-end lg:justify-center z-1 md:pr-10 lg:pr-0"
      >
        <div
          className="transform md:skew-x-6 self-center max-w-md p-5"
        >
          <h1
            className="text-4xl font-bold text-gray-50"
          >
            ANDY'S AMERICAN
            <br />
            KUSTOMS
          </h1>
          <p
            className="text-gray-200 font-light my-5"
          >
            This is my one man business to restore, repair or modify just about anything! Diagnostics using oscilloscopes, scan tools and test lights are skills that I continue to work on everyday. But my main passion is bringing old machinery back to life and perhaps injecting some new technology at the same time!! Have a look at my projects below and contact me if you think I can help.
          </p>
          <a
            href="/projects"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            <button
              className="px-6 py-2.5 text-gray-100 transition-colors duration-150 font-medium rounded-tl-lg rounded-br-lg bg-gray-700 border-2 border-gray-700 focus:shadow-outline hover:bg-gray-800 hover:border-gray-800 mr-3"
            >
              View Projects
            </button>
          </a>
          <a
            href="/contact"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            <button
              className="inline-flex items-center px-6 py-2.5 text-gray-300 transition-colors duration-150 font-medium rounded-md border-gray-300 focus:shadow-outline hover:border-gray-400 hover:text-gray-400"
            >
              <span>
                Contact
              </span>
              <svg
                className="w-6 h-4 ml-2"
                fill="currentColor"
                viewBox="0 0 20 20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clipRule="evenodd"
                  d="M12.293 5.293a1 1 0 011.414 0l4 4a1 1 0 010 1.414l-4 4a1 1 0 01-1.414-1.414L14.586 11H3a1 1 0 110-2h11.586l-2.293-2.293a1 1 0 010-1.414z"
                  fillRule="evenodd"
                />
              </svg>
            </button>
          </a>
        </div>
      </div>
      <div
        className="right bg-hero-bg bg-no-repeat sm:w-1/3 md:w-2/3 -ml-20 object-cover h-full bg-cover bg-right hidden sm:flex"
      >
        <div
          className="absolute h-2/3 w-36 transform -skew-x-6 bg-secondary self-center"
        />
      </div>
    </div>
    <div
      className="epic-gray h-500"
    >
      <div
        className="mx-auto container md:flex"
      >
        <p
          className="text-2xl font-light pt-20 md:w-3/4"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras maximus mi nec sem sodales, eget pellentesque neque pulvinar. Phasellus sollicitudin erat sit amet tempus gravida. Nam in ex molestie
        </p>
        <div
          className="pt-10 md:pt-20 flex-1 text-center self-center"
        >
          <a
            href="/projects"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            <button
              className="text-grey-800 rounded-tl-lg rounded-br-lg border-2 border-gray-800 text-lg py-3 px-5 inline-flex items-center"
            >
              <span>
                All Projects
              </span>
              <svg
                className="w-6 h-4 ml-2"
                fill="currentColor"
                viewBox="0 0 20 20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clipRule="evenodd"
                  d="M12.293 5.293a1 1 0 011.414 0l4 4a1 1 0 010 1.414l-4 4a1 1 0 01-1.414-1.414L14.586 11H3a1 1 0 110-2h11.586l-2.293-2.293a1 1 0 010-1.414z"
                  fillRule="evenodd"
                />
              </svg>
            </button>
          </a>
        </div>
      </div>
    </div>
    <div
      className="md:-mt-60 w-full"
    >
      <div
        className="container mx-auto"
      >
        <div
          className="grid md:grid-cols-3 sm:grid-cols-2 gap-2"
        >
          <div
            className="relative"
          >
            <a
              href="/projects/wow"
              onClick={[Function]}
              onMouseEnter={[Function]}
            >
              <div
                style={
                  Object {
                    "boxSizing": "border-box",
                    "display": "block",
                    "margin": 0,
                    "overflow": "hidden",
                    "position": "relative",
                  }
                }
              >
                <div
                  style={
                    Object {
                      "boxSizing": "border-box",
                      "display": "block",
                      "paddingTop": "100%",
                    }
                  }
                />
                <img
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style={
                    Object {
                      "border": "none",
                      "bottom": 0,
                      "boxSizing": "border-box",
                      "display": "block",
                      "height": 0,
                      "left": 0,
                      "margin": "auto",
                      "maxHeight": "100%",
                      "maxWidth": "100%",
                      "minHeight": "100%",
                      "minWidth": "100%",
                      "objectFit": "cover",
                      "objectPosition": undefined,
                      "padding": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                      "visibility": "hidden",
                      "width": 0,
                    }
                  }
                />
              </div>
              <div
                className="false opacity-0 absolute flex top-0 left-0 p-2 w-full h-full justify-center cursor-pointer text-gray-50 text-2xl items-center bg-black bg-opacity-50 transition ease-in-out duration-200"
                onFocus={[Function]}
                onMouseLeave={[Function]}
                onMouseOver={[Function]}
              >
                <span
                  className="border-solid border-2 p-4 md:p-8 text-center border-gray-50"
                >
                  First Post!!
                </span>
              </div>
            </a>
          </div>
        </div>
      </div>
    </div>
  </main>,
  <div
    className="container mx-auto mt-16"
  >
    <div
      className="flex place-content-center py-8"
    >
      <a
        href="https://www.facebook.com/andyskustoms/"
      >
        <img
          alt="facebook icon"
          className="mx-1"
          height={40}
          src="test-file-stub"
          width={40}
        />
      </a>
      <a
        href="https://www.instagram.com/andysamericankustoms/"
      >
        <img
          alt="instagram icon"
          className="mx-1"
          height={40}
          src="test-file-stub"
          width={40}
        />
      </a>
    </div>
  </div>,
]
`;
